src/energyplus/ForwardTranslator/ForwardTranslateZoneHVACTerminalUnitVariableRefrigerantFlow.cpp

```diff
@@ -264,121 +249,138 @@ namespace energyplus {
       // OutsideAirMixerObjectName
 
       idfObject.setString(ZoneHVAC_TerminalUnit_VariableRefrigerantFlowFields::OutsideAirMixerObjectName, _outdoorAirMixer.name().get());
-    }
 
-    if (coolingCoil) {
-      if (boost::optional<IdfObject> _coolingCoil = translateAndMapModelObject(coolingCoil.get())) {
-        // CoolingCoilObjectType
+      // For the remaining components, the first node is the mixerOutletNodeName
+      inletNodeName = mixerOutletNodeName;
+    }
 
-        idfObject.setString(ZoneHVAC_TerminalUnit_VariableRefrigerantFlowFields::CoolingCoilObjectType, _coolingCoil->iddObject().name());
+    struct Component
+    {
+      Component(const IdfObject& obj, const std::string t_outletNodeName, unsigned t_inletIndex, unsigned t_outletIndex)
+        : idfObject(obj), outletNodeName(t_outletNodeName), inletIndex(t_inletIndex), outletIndex(t_outletIndex) {}
+      IdfObject idfObject;
+      std::string outletNodeName;
+      unsigned inletIndex;
+      unsigned outletIndex;
+    };
 
-        // CoolingCoilObjectName
+    size_t fanIndex = -1;
 
-        idfObject.setString(ZoneHVAC_TerminalUnit_VariableRefrigerantFlowFields::CoolingCoilObjectName, _coolingCoil->name().get());
+    // We'll store the Components in the vector, so we can write their inlet/outlet nodes in the right order
+    std::vector<Component> compsInOrder;
```

Comment:
Ok, so I create a vector to store any components after the Mixer (if any), I fill the vector in the order of the components, and once we're done populating the vector, I can write the nodes without making any mistakes.